rocAL/rocAL/source/pipeline/ring_buffer.cpp

```diff
@@ -245,11 +260,47 @@ void RingBuffer::initBoxEncoderMetaData(RocalMemType mem_type, size_t encoded_bb
     }
 #else    
     {
-        //todo:: for host
+        if(_meta_data_sub_buffer_count < 2)
+            THROW("Insufficient HOST metadata buffers for Box Encoder");
+        // Check if sufficient data has been allocated for the labels and bbox host buffers if not reallocate
+        for(size_t buffIdx = 0; buffIdx < BUFF_DEPTH; buffIdx++)
```

Comment:
the entire code for device above was specifically added for box encoder data which is fixed. So I am not sure what the CPU code is doing below. How is the reallocation handled for device case.

Reply:
For the device case the previous flow would be followed. In case of HOST backend we are returning the HOST buffers from here directly and reallocating if required,